api/server/router/system/system_routes.go

```diff
@@ -43,6 +47,24 @@ func (s *systemRouter) pingHandler(ctx context.Context, w http.ResponseWriter, r
 	return err
 }
 
+func (s *systemRouter) swarmState() string {
+	if s.cluster == nil {
+		return string(swarm.LocalNodeStateInactive)
+	}
+
+	swarmState := string(swarm.LocalNodeStateInactive)
+	ci := s.cluster.Info()
```

Comment:
This seems expensive to call for `ping`, which is easily called multiple times by the official client for a single action.

Reply:
Yes, I was thinking about that as well, should probably have left a comment about that 🤔 

I can look around if there's possibly a more lightweight approach to get the needed info (unless you have suggestions)

Reply:
A dedicated function to get just what we need seems ideal.
Specifically for stuff like this that can change and as such requires locking I would probably set a value much like we do for the container state counters for `docker info`.